internal/pointer/bool.go

```diff
@@ -0,0 +1,13 @@
+package pointer
+
+func BoolPtr(b bool) *bool {
+	boolVar := b
+	return &boolVar
+}
+
+func CompareBoolPtr(ptr *bool, b bool) bool {
+	if ptr == nil {
```

Comment:
Pity that nil is possible at all here, is it possible to eliminate nil, e.g. the null object pattern?

I would have expected that nil means absent, which is typically default false.  Condition in typescript would be something like ((ptr ?? false) == b).

Reply:
So here I've added a helper function for comparing bool ptrs and bools. If we try to do a compare of an empty ptr and a bool we'll get panic nil pointer. So we always have to do the nil check. 

Rather than do this in line every time we need to compare bool and ptr I've added this function.